Fix Guest Auth

Yicong-Huang opened the pull request:
Fixes #1391 by introducing a new `SessionRole` called `Guest`. 

Now backend can be configured to enable the user system or not. By default, the user system is disabled, and the `GUEST` role can be used as a placeholder for endpoints that optionally require user information, such as `api/queryplan/autocomplete`.

The original user role is now changed to `BASIC` to be differentiated from `GUEST`.

zuozhiw reviewed (approved):
Looks good.

Since we have added one more conf to enable user dashboard, can you update our wiki to include this config: https://github.com/Texera/texera/wiki/Guide-for-Developers#to-enable-user-dashboard

Also the guest account is available when the user dashboard is not enabled. I'm thinking if there is a way we automatically assign the guest account to non-logged-in user. 
For example, visit the production system without logging in.

Yicong-Huang commented:
> Looks good.
> 
> Since we have added one more conf to enable user dashboard, can you update our wiki to include this config: https://github.com/Texera/texera/wiki/Guide-for-Developers#to-enable-user-dashboard
> 
> Also the guest account is available when the user dashboard is not enabled. I'm thinking if there is a way we automatically assign the guest account to non-logged-in user. For example, visit the production system without logging in.

added.
